src/shogun/base/SGObject.cpp

```diff
@@ -691,19 +651,20 @@ void CSGObject::build_gradient_parameter_dictionary(CMap<TParameter*, CSGObject*
 	}
 }
 
-CSGObject* CSGObject::clone() const
+template <typename M>
+typename CSGObjectBase<M>::Derived* CSGObjectBase<M>::clone() const
 {
 	SG_DEBUG("Starting to clone %s at %p.\n", get_name(), this);
 	SG_DEBUG("Constructing an empty instance of %s.\n", get_name());
-	CSGObject* clone = create_empty();
+	Derived* clone = create_empty();
```

Comment:
you can auto this (unless you used a regex/sed :P)

Reply:
you caught me :"D

Reply:
k then for the time being leave it... will change it with a libtooling script

Reply:
don't worry about it, i'll change them, they are only a couple of instances